lib/reduce_quda.cu

```diff
@@ -612,6 +638,33 @@ namespace quda {
 #else
             errorQuda("QUDA_PRECISION=%d does not enable precision %d", QUDA_PRECISION, x.Precision());
 #endif
+
+          } else if (x.Precision() == QUDA_QUARTER_PRECISION) {
+#if QUDA_PRECISION & 1
+            if (x.Nspin() == 4) { // wilson
+#if defined(GPU_WILSON_DIRAC) || defined(GPU_DOMAIN_WALL_DIRAC)
+              const int M = 6;
```

Comment:
just by looking at the diff it seems odd that M=6 here and all other changed lines have M=12 (M=3 for staggered). But admittedly I did not look at it with context yet.